src/string.cr

```diff
@@ -563,13 +563,18 @@ class String
     !!index(str)
   end
 
-  def split
+  def split(limit = nil : Int32?)
+    if limit && limit <= 1
```

Comment:
What about using the inline `if` for guard statements? I find it to read better. In this case:

```crystal
def split(limit = nil : Int32?)
  return [self] if limit && limit <= 1
  ...
end
```